### PR TITLE
Glcontext cleanups

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: libnodegl/backends/gl/glcontext.c [hunk 1]
```diff
@@ -610,26 +610,32 @@ int ngli_glcontext_check_extension(const char *extension, const char *extensions
...
 
-    LOG(ERROR, "%s: GL error: %s (0x%04x)", context, error_str, error);
+        LOG(ERROR, "%s: GL error: %s (0x%04x)", context, error_str, error);
+        ret = -1;
+    }
 
 #if DEBUG_GL
-    ngli_assert(0);
+    if (ret < 0)
+        ngli_assert(0);
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/glcontext.c:637]: `ngli_assert(ret >= 0)`?

File: libnodegl/backends/gl/glcontext.c [hunk 2]
```diff
@@ -610,26 +610,32 @@ int ngli_glcontext_check_extension(const char *extension, const char *extensions
...
-    GL_ERROR_STR_CASE(GL_OUT_OF_MEMORY);
-    default:
-        error_str = "unknown error";
-    }
+    int ret = 0;
+
+    while (1) {
+        const GLenum error = ngli_glGetError(glcontext);
+        if (!error)
+            break;
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/glcontext.c:618]: you may want to break if the error is the same as the previous loop to avoid an infinite loop

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]

File: libnodegl/backends/gl/glcontext.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/glcontext.c:618]: Switched to a for loop.

File: libnodegl/backends/gl/glcontext.c (hunk 1 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/glcontext.c]: Done.

File: libnodegl/backends/gl/glcontext.c [hunk 3]
```diff
@@ -606,40 +606,35 @@ int ngli_glcontext_check_extension(const char *extension, const char *extensions
...
+        GL_ERROR_STR_CASE(GL_INVALID_VALUE);
+        GL_ERROR_STR_CASE(GL_INVALID_OPERATION);
+        GL_ERROR_STR_CASE(GL_INVALID_FRAMEBUFFER_OPERATION);
+        GL_ERROR_STR_CASE(GL_OUT_OF_MEMORY);
+        default:
+            error_str = "unknown error";
+        }
+
+        LOG(ERROR, "%s: GL error: %s (0x%04x)", context, error_str, error);
+        ret = -1;
```
Comment (cboesch-gpsw) [inline libnodegl/backends/gl/glcontext.c:632]: you may want to `return NGL_ERROR_GRAPHICS_GENERIC` and do something prettier in `gpu_ctx_gl.c` (not a blocking comment)

**Review by cboesch-gpsw — APPROVED** [review]

File: libnodegl/backends/gl/glcontext.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/backends/gl/glcontext.c]: Done.

**Pull request merged**